sdk/identity/azure-identity/src/client_certificate_credential.cpp

```diff
@@ -399,16 +400,33 @@ ClientCertificateCredential::ClientCertificateCredential(
     CertificateThumbprint mdVec;
     try
     {
+      if (clientCertificatePath.empty())
+      {
+        throw AuthenticationException("Certificate file path is empty.");
+      }
```

Comment:
Why are we checking and throwing this within a try/catch? Shouldn't this input parameter validation go above?

Reply:
Why not, and plus try-catch block is going to prepend `Identity: ClientCertificateCredential` to it.